state.go

```diff
@@ -397,6 +397,16 @@ func NewUpdateErrorState(err menderError, update UpdateResponse) State {
 func (ue *UpdateErrorState) Handle(c Controller) (State, bool) {
 	// TODO error handling
 	c.ReportUpdateStatus(ue.update, statusFailure)
+
+	// TODO fetch logs from logger
+	c.UploadLog(ue.update, []LogEntry{
+		LogEntry{
+			Timestamp: time.Now().Format(time.RFC3339),
+			Level:     "error",
+			Message:   "update failed",
+		},
+	})
```

Comment:
I think it's still a fairly technical discussion, we need to make sure we can always recover. So if we break it down:

* UpdateErrorState should never be reached in a daemon running with a non-committed update, because being in that situation means that there will be an immediate reboot and most likely rollback.
* After rolling back, and we are back in the old update, we enter UpdateErrorState, and stay there until we have successfully reported it.
* If there's a reboot while in UpdateErrorState (not necessarily because of us), we should stay in UpdateErrorState, IOW it should have no influence on the state. Only the report will change the state.

How to implement this technically is another matter, I'm just wondering if you agree with my logical rationale.